src/dra/42398.c

```diff
@@ -227,7 +227,20 @@ void func_800E376C(void) {
 // mips to c doesn't support this function very well
 INCLUDE_ASM("asm/dra/nonmatchings/42398", func_800E385C);
 
-INCLUDE_ASM("asm/dra/nonmatchings/42398", func_800E38CC);
+void func_800E38CC(void) {
+    if (D_800A015C != 0) {
+        if ((u32)D_800A0158 >= 0x24U) {
+            D_801390D4->buf.disp.screen.x = 0;
+            PutDispEnv(&D_801390D4->buf.disp);
+            return;
```

Comment:
You can get rid of this return by using an `else`. I believe that is how developers wrote the real code.